enhancements/metadata/user-tag-management.md

```diff
@@ -0,0 +1,372 @@
+from the cloud resources.
+
+### High-level workflow
+
+#### Bootstrap
+1. User starts the controller with cloud service provider credentials and sync period set.
+2. Controller starts to list/watch custom resource.
+3. User creates `AWSMetadata` object with tag list details.
+4. User creates `CloudMetadata` configuration with controller configuration.
+5. Based on the configuration, controller queries and lists all resources based on classifier tag.
+6. Creates list of existing tags on cloud resource.
+7. Add/update tag workflow is initiated.
+
+#### Add/Update tags
+
+1. User creates `AWSMetadata` object with tag list details.
+2. User creates `CloudMetadata` configuration with controller configuration.
+3. Controller updates ready condition to false in `MetadataStatus.status`.
+4. Controller validates tag entries from `AWSMetadata.spec.resourcetags` and populates a new list of tags to be added to cloud resource.
+5. Controller trims the tag list as per maximum limit policy specific to cloud resource.
+6. Controller performs lexicographic sorting.
+7. Controller replaces the tag list on the cloud resource.
+8. Controller updates the tag list to `AWSMetadata.status`.
+9. Controller updates ready condition to true in `MetadataStatus.status`.
+
+#### Delete tags
+
+1. User adds tag key list (optionally, value, for validation) to custom resource specification.
+2. User specifies operation to be performed on the tag list by specific opcode.
+3. Controller updates ready condition to false in `MetadataStatus.status`.
+4. Controller gets the active tag list, removes requested tags from the list and populates a new list of tags.
+5. Controller replaces the tag list on the cloud resource.
+6. Controller updates the tag list to `AWSMetadata.status`.
+7. Controller updates ready condition to true in `MetadataStatus.status`.
+
+#### Handling of pre-existing tags or tags added using other tools
+
+In case of add operation, controller appends tags to existing tag list. If there is any tag pre-existing with same key name,
+there will be an error reported for the operation. While, in case of update operation, the pre-existing tag is updated. There is
+no distinction made between pre-existing tags and tags in `AWSMetadataSpec` for reconciliation. The active tag list is listed in the status.
+
+#### Classifier tag behaviour
+
+Classifier tags are used to enable controller to get applicable cloud resources to manage tags. Classifier tags are mandatory and should
+be added to cloud resource. Controller does not support any wildcard characters in classifier strings.
+When multiple classifiers are used, a logical OR condition is applied for the classifiers.
+
+#### Reconciliation
+
+Reconciliation of tags is ignored when ready condition is set to false at `MetadataStatus.status`. Reconciliation of tags is based on active tag list in `AWSMetadata.status.resourcetags`. On deletion of AWSMetadataRef
```

Comment:
metadatastatus.status on what resource?  and when would that ready condition be set to false?

Below you say it would be set to false when there's no AWSMetadataRef, but i don't think that's what i'd expect....the controller is still ready(assuming that's what the condition refers to), it just has nothing to do.